### PR TITLE
Feature/member edit ux

### PR DESCRIPTION
**Pull request by TwistMeister**
## Changes
_Please describe the essence of this PR in a few sentences._

## Issue ticket number and link
https://github.com/minvws/nl-kat-coordination/issues/584

## Proof
Screenshot 2023-04-06 at 17 33 01

## Checklist for author(s):
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR comes from a `feature` or `hotfix` branch, in line with our git branching strategy;
- [x] This PR is "bite-sized" and only focuses on a single issue, problem, or feature;
- [x] I am not reinventing the wheel: there is no high-quality library that already has this feature;
- [ ] I have changed the example `.env` files if I added, removed, or changed any config options, and I have informed others that they need to modify their `.env` files if required;
- [ ] I have performed a self-review of my own code;
- [x] I have commented my code, particularly in hard-to-understand areas;
- [ ] I have made corresponding changes to the documentation, if necessary;
- [x] I have written unit, integration, and end-to-end tests for the change that I made;

If a non-trivial PR:
- [x] This PR is part of a milestone and has appropriate labels;
- [x] This PR is properly linked to the project board (either directly or via an issue);
- [x] I have added screenshots or some other proof that my code does what it is supposed to do;

```
## Checklist for functional reviewer(s):
- [ ] If a non-trivial PR: This PR is properly linked to an issue on the project board;
- [ ] I have checked out this branch, and successfully ran `make kat`;
- [ ] I have ran `make test-rf` and all end-to-end Robot Framework tests pass;
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended;
- [ ] I confirmed that there are no unintended functional regressions in this branch;

### What works:
* _bullet point + screenshot (if useful) per tested functionality_

### What doesn't work:
* _bullet point + screenshot (if useful) per tested functionality_

### Bug or feature?:
* _bullet point + screenshot (if useful) if it is unclear whether something is a bug or an intended feature._
```

```
## Checklist for code reviewer(s):
- [ ] The code passes the CI tests and linters;
- [ ] The code does not bypass authentication or security mechanisms;
- [ ] The code does not introduce any dependency on a library that has not been properly vetted;
- [ ] The code does not violate Model-View-Template and our other architectural principles;
- [ ] The code contains docstrings, comments, and documentation where needed;
- [ ] The code prioritizes readability over performance where appropriate;
- [ ] The code conforms to our agreed coding standards.
```

### CODE REVIEW

File: rocky/tools/models.py [hunk 1]
```diff
@@ -155,13 +155,13 @@ class OrganizationMember(models.Model):
     class STATUSES(models.TextChoices):
```
Comment (praseodym) [inline rocky/tools/models.py:155]: Do we still need this? What's the difference between 'active' and 'new'?

File: rocky/account/models.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def organizations(self) -> List[Organization]:
             return self.all_organizations
         return [
             m.organization
-            for m in filter(lambda o: o.status is not OrganizationMember.STATUSES.BLOCKED, self.organization_members)
+            for m in filter(lambda o: o.status is not OrganizationMember.blocked, self.organization_members)
```
Comment (praseodym) [inline rocky/account/models.py:110]: This shouldn't check `o.status` but just `o.blocked`

**Review by praseodym — CHANGES_REQUESTED** [review]

File: rocky/tools/models.py (hunk 1 above)
Comment (praseodym) [reply rocky/tools/models.py:155]: I see! Maybe this should also be a Boolean value instead?

Comment (TwistMeister) [reply rocky/tools/models.py:152]: That could be, but not necessarily imo. I can't foresee if there will be more statuses added in the future. My suggestion would be to leave this as is for now.

File: rocky/account/models.py (hunk 2 above)
Comment (TwistMeister) [reply rocky/account/models.py]: Good catch, fixed it 👍

File: rocky/tools/models.py (hunk 1 above)
Comment (praseodym) [reply rocky/tools/models.py:152]: Ok!

File: yarn.lock [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
[the remark this answers is not in the document]
Comment (TwistMeister) [reply yarn.lock]: Deleted

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make kat`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* Checking/unchecking Blocked works correctly as a superadmin
* Superadmins are not editable/blockable
* Other users also cannot edit/block superadmins
* Admins cannot block themselves through the Edit page

### What doesn't work:
* As a regular admin, Blocking someone from the /members page (not the edit page) does not appear to be applied

### Bug or feature?:
* Admins can block themselves, leaving them with a 403 message. However, refreshing the page unblocks them. 
* No green confirmation message when blocking/unblocking a user through the /members page

**Review by praseodym — DISMISSED** [review]

**Comment by TwistMeister** [comment]
@Darwinkel
Fixed both cases, could you please verify if it works properly? Admins can't block themself anymore and can block other users without getting the 403 they got earliers.
- As a regular admin, Blocking someone from the /members page (not the edit page) does not appear to be applied
- Admins can block themselves, leaving them with a 403 message. However, refreshing the page unblocks them.

**Comment by TwistMeister** [comment]
@Darwinkel Added the green confirmation message when blocking/unblocking a user through the /members page

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make kat`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* Green confirmation message has been added when blocking/unblocking a user through the /members page
* Admins can't block themselves anymore
* Admins no longer get a 403 themselves

### What doesn't work:
* Admins can block each other from the /members page, but not the edit form.

**Comment by TwistMeister** [comment]
@Darwinkel Fixed the bug where admins couldn't be blocked on the edit form.

**Review by Darwinkel — DISMISSED** [review]

**Review by dekkers — CHANGES_REQUESTED** [review]

File: rocky/account/forms/account_setup.py [hunk 4]
```diff
@@ -172,18 +172,28 @@ class Meta:
...
+        widget=forms.CheckboxInput(),
+    )
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-
+        self.fields["blocked"].widget.attrs["field_form_label"] = "Status"
+        if self.instance.user.is_superuser:
+            self.fields["trusted_clearance_level"].disabled = True
+        self.fields["acknowledged_clearance_level"].label = _("Accepted clearance level")
```
[the remark this answers is not in the document]
Comment (TwistMeister) [reply rocky/account/forms/account_setup.py:196]: For better understanding for the user. But you are right, we use "agreed clearance level" in the member list and here we use "accepted". I asked @HeleenSG which of the two we will want to stick to and make changes accordingly.

**Review by dekkers — APPROVED** [review]

**Review by dekkers — APPROVED** [review]

**Pull request merged**